### DISCUSSION TITLE
Coordinate fields for Locations

### DISCUSSION THREAD

**Discussion by Hyperopic** [netbox-community/netbox#10765] [Ideas]
It would be useful to have the Latitude & Longitude fields present in Location objects as well as in Sites.
We have a site which is a large open campus and there are several outdoor locations (cabinets and chambers) that have splice closures and GPON splitters. At the moment I'm modelling these as Location objects with GPS coordinates in Description, but this lacks validation and becomes awkward when trying to extract these when additional information has been entered in Description.
I did look at using Sites but this becomes messy when using Sites elsewhere to describe a single building site which has no need for locations within the site. The chambers\cabinets also don't need to have all the associated address records since they are subsets of the Site

**Comment by Kludge-As-A-Service** [comment]
Have you considered custom fields?

**Comment by eronlloyd** [comment]
I, too, manage a large OSP with a PON service deployed. We use sites for any point where cable terminations happen, whether it's a building or a hand hole with a splice closure in it. Locations are within sites at the same coordinates, and fed by the same ISP links, such as floors, units, common spaces, distribution rooms, etc.

To keep it simple, the rule of thumb is:
- Sites are any point where OSP terminations occur
- Locations are any point inside a site where ISP terminations occur

I'm also using a GIS to spatially document all this, which helps a great deal, and am experimenting with ways to share the site data as well as make Netbox more spatially aware. I'll keep you in mind when I'm ready to share progress on it.

**Comment by vishalmalli** [comment]
This feature would be beneficial for my use case as well. We consider a site to be anything that has its own physical address. Some of our sites have multiple buildings, which we add as locations in NetBox. The coordinate field for locations would be really nice so our NMS can accurately map the locations of all our equipment.

**Comment by eronlloyd** [comment]
I'm thinking of a way to implement global spatial functionality by creating a custom JSON field to store a GeoJSON record, which could be created/edited in a Leaflet view. From there, We could store not only coordinates but even base maps for regions and perhaps even polygons where applicable. Further manipulation could happen using [Shapely](https://github.com/shapely/shapely).

Is there interest in this? For my needs, we're moving lots of telecom OSS into Netbox, and having map views of records like sites is important. It would be awesome if Netbox implemented GeoDjango under the hood to fully enable spatial functionality, but this could be a starting point.

**Comment by mtinberg** [comment]
If you look in github and search for netbox-plugins, there already is one using leaflet and a custom field on Device to store coordinates to display Devices on a map.
https://github.com/drygdryg/netbox-plugin-device-map

This might be a good starting place, or maybe it isn't the kind of thing you are looking for, but working on a plugin to add this kind of feature makes a lot of sense


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Reply by eronlloyd** [reply]
Thanks @mtinberg! I missed that in my review. I'm aiming for a more general purpose plugin, starting with sites, but I'll take a closer look. Cheers!
